assets/app/scripts/controllers/deployments.js

```diff
@@ -8,34 +8,44 @@
  * Controller of the openshiftConsole
  */
 angular.module('openshiftConsole')
-  .controller('DeploymentsController', function ($scope, DataService, $filter, LabelFilter, Logger) {
+  .controller('DeploymentsController', function ($scope, DataService, $filter, LabelFilter, Logger, ImageStreamResolver) {
     $scope.deployments = {};
+    $scope.podTemplates = {};
     $scope.unfilteredDeployments = {};
-    $scope.images = {};
+    $scope.imageStreams = {};
     $scope.imagesByDockerReference = {};
+    $scope.imageStreamImageRefByDockerReference = {}; // lets us determine if a particular container's docker image reference belongs to an imageStream
     $scope.builds = {};
     $scope.labelSuggestions = {};
     $scope.alerts = $scope.alerts || {};    
     $scope.emptyMessage = "Loading...";
     var watches = [];
 
+    var podTemplates = function() {
+      angular.forEach($scope.deployments, function(deployment, deploymentId){
+        $scope.podTemplates[deploymentId] = deployment.spec.template;
+      });
+    };
+
     watches.push(DataService.watch("replicationcontrollers", $scope, function(deployments) {
       $scope.unfilteredDeployments = deployments.by("metadata.name");
       LabelFilter.addLabelSuggestionsFromResources($scope.unfilteredDeployments, $scope.labelSuggestions);
       LabelFilter.setLabelSuggestions($scope.labelSuggestions);
       $scope.deployments = LabelFilter.getLabelSelector().select($scope.unfilteredDeployments);
+      podTemplates();
```

Comment:
Just a nit... it's not very clear what podTemplates() is doing from the name... maybe something like:
```
$scope.podTemplates = generatePodTemplates($scope.deployments)
```
would be more readable